core/src/test/java/com/linecorp/armeria/server/InitiateConnectionShutdownTest.java

```diff
@@ -0,0 +1,350 @@
+import io.netty.handler.codec.http2.DefaultHttp2Headers;
+import io.netty.handler.codec.http2.Http2ConnectionHandler;
+import io.netty.handler.codec.http2.Http2ConnectionHandlerBuilder;
+import io.netty.handler.codec.http2.Http2ConnectionPrefaceAndSettingsFrameWrittenEvent;
+import io.netty.handler.codec.http2.Http2Error;
+import io.netty.handler.codec.http2.Http2FrameListener;
+import io.netty.handler.codec.http2.Http2Headers;
+import io.netty.handler.logging.LoggingHandler;
+import io.netty.util.AsciiString;
+
+class InitiateConnectionShutdownTest {
+
+    @RegisterExtension
+    static final ServerExtension goAwayServer = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            sb.annotatedService(new AnnotatedTestService());
+            sb.connectionDrainDurationMicros(0);
+        }
+    };
+    @RegisterExtension
+    static final ServerExtension goAwayServerNoopKeepAliveHandler = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            sb.annotatedService(new AnnotatedTestService());
+            sb.idleTimeoutMillis(0);
+            sb.requestTimeoutMillis(0);
+            sb.maxConnectionAgeMillis(0);
+            sb.maxNumRequestsPerConnection(0);
+            sb.connectionDrainDurationMicros(0);
+        }
+    };
+    private static final int STREAM_ID = 3;
+    private static final int PADDING = 0;
+    private static final ByteBuf DEBUG_DATA = Unpooled.unreleasableBuffer(
+            Unpooled.copiedBuffer("app-requested", StandardCharsets.UTF_8));
+    private static final AtomicBoolean connectionClosed = new AtomicBoolean();
+    @Mock
+    private Http2FrameListener clientListener;
+    private Http2ConnectionHandler http2Client;
+    private Channel clientChannel;
+    private Cleaner cleaner;
+
+    private static Http2Headers getHttp2Headers(String path) {
+        return new DefaultHttp2Headers(false)
+                .method(new AsciiString("GET"))
+                .scheme(new AsciiString("http"))
+                .path(new AsciiString(path));
+    }
+
+    private void makeHttp2Request(String path) throws Exception {
+        final AtomicBoolean finished = new AtomicBoolean();
+        when(clientListener.onDataRead(any(), anyInt(), any(), anyInt(), anyBoolean())).thenAnswer(
+                invocation -> {
+                    finished.set(true);
+                    return 0;
+                });
+        doAnswer((Answer<Void>) invocation -> {
+            // Retain buffer for comparison in tests.
+            final ByteBuf buf = invocation.getArgument(3);
+            buf.retain();
+            // Cleanup after test finish.
+            cleaner.add(buf::release);
+            return null;
+        }).when(clientListener).onGoAwayRead(any(ChannelHandlerContext.class), anyInt(), anyLong(),
+                                             any(ByteBuf.class));
```

Comment:
Could we use `when().thenAnswer()` or `doAnswer().when()` consistently rather than mixing the two styles?